language/tools/read-write-set/src/lib.rs

```diff
@@ -71,136 +71,39 @@ impl ReadWriteSetAnalysis {
-    /// by `module::fun` if called with arguments `signers`, `actuals`, `type_actuals` in state
-    /// `blockchain_view`.
-    pub fn get_keys_written(
-        &self,
-        module: &ModuleId,
-        fun: &IdentStr,
-        signers: &[AccountAddress],
-        actuals: &[Vec<u8>],
-        type_actuals: &[TypeTag],
-        blockchain_view: &impl MoveResolver,
-    ) -> Result<Vec<ResourceKey>> {
-        self.get_concretized_keys(
-            module,
-            fun,
-            signers,
-            actuals,
-            type_actuals,
-            blockchain_view,
-            true,
-        )
-    }
-
-    /// Returns an overapproximation of the `ResourceKey`'s in global storage that will be read by
-    /// `module::fun` if called with arguments `signers`, `actuals`, `type_actuals` in state
-    /// `blockchain_view`.
-    pub fn get_keys_read(
-        &self,
-        module: &ModuleId,
-        fun: &IdentStr,
-        signers: &[AccountAddress],
-        actuals: &[Vec<u8>],
-        type_actuals: &[TypeTag],
-        blockchain_view: &impl MoveResolver,
-    ) -> Result<Vec<ResourceKey>> {
-        self.get_concretized_keys(
-            module,
-            fun,
-            signers,
-            actuals,
-            type_actuals,
-            blockchain_view,
-            false,
-        )
-    }
-
-    /// Returns an overapproximation of the `ResourceKey`'s in global storage that will be accesses
-    /// by module::fun` if called with arguments `signers`, `actuals`, `type_actuals` in state
-    /// `blockchain_view`.
-    /// If `is_write` is true, only ResourceKey's written will be returned; otherwise, only
-    /// ResourceKey's read will be returned.
-    pub fn get_concretized_keys(
-        &self,
-        module: &ModuleId,
-        fun: &IdentStr,
-        signers: &[AccountAddress],
-        actuals: &[Vec<u8>],
-        type_actuals: &[TypeTag],
-        blockchain_view: &impl MoveResolver,
-        is_write: bool,
-    ) -> Result<Vec<ResourceKey>> {
-        if let Some(state) = self.get_summary(module, fun) {
-            let results = dynamic_analysis::concretize(
-                state.accesses().clone(),
-                signers,
-                actuals,
-                type_actuals,
-                &self.get_function_env(module, fun).unwrap(),
-                blockchain_view,
-            )?;
-            Ok(if is_write {
-                results.get_keys_written(&self.env)
-            } else {
-                results.get_keys_read(&self.env)
-            })
-        } else {
-            bail!("Couldn't resolve function {:?}::{:?}", module, fun)
-        }
-    }
-
     /// Returns the FunctionEnv for `module`::`fun`
     /// Returns `None` if this function does not exist
     pub fn get_function_env(&self, module: &ModuleId, fun: &IdentStr) -> Option<FunctionEnv> {
         self.env
             .find_function_by_language_storage_id_name(module, &fun.to_owned())
     }
+
+    pub fn normalize(
```

Comment:
It might make sense to call this `normalize_all_scripts` or something to convey that it's iterating over all modules + functions and *not* just normalizing `add_ons`.